OpenRA.Mods.Common/Widgets/Logic/Editor/ActorSelectorLogic.cs

```diff
@@ -94,6 +86,22 @@ public ActorSelectorLogic(Widget widget, World world, WorldRenderer worldRendere
 				ownersDropDown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 270, owners, setupItem);
 			};
 
+			editorLayer.OnSpawnRemoved = () =>
+			{
+				if (editorLayer.Players.Players.Values.Any(p => p.Name.Equals(selectedOwner.Name)))
+					return;
+
+				var orderedOwners = editorLayer.Players.Players.Values.ToList();
```

Comment:
It should be possible to simplify this down to something like the following (but note that I haven't tested this, so you may need to tweak it):
```csharp
// Fall back to the previous owner in the list.
var previousOwners = editorLayer.Players.Players.Values
	.Append(selectedOwner)
	.OrderBy(x => x.Name)
	.TakeWhile(x => x != selectedOwner);

SelectOwner(previousOwners.LastOrDefault() ?? previousOwners.First());
```